torch/testing/_internal/common_fsdp.py

```diff
@@ -104,28 +187,31 @@ def __init__(self, wrap_fsdp, cpu_offload=CPUOffload(offload_params=False)):
         super().__init__()
         # keep everything deterministic for model initialization
         torch.manual_seed(0)
-        self.inner: Union[torch.nn.Linear, FullyShardedDataParallel] = \
+        self.inner: Union[torch.nn.Linear, FSDP] = \
             torch.nn.Linear(2, 2).cuda()
         if wrap_fsdp:
-            self.inner = FullyShardedDataParallel(self.inner, cpu_offload=cpu_offload)
+            self.inner = FSDP(self.inner, cpu_offload=cpu_offload)
         self.outer = torch.nn.Linear(2, 2).cuda()
 
     def forward(self, x):
         y = self.inner(x)
         return self.outer(y)
 
-class TransformerWithSharedParams(nn.Module):
+class TransformerWithSharedParams(FSDPTestModel):
     def __init__(
-        self, group, *args, d_vocab=23, d_model=16, add_bn=True,
-        fsdp_init_mode=FSDPInitMode.CUDA_AFTER, **kwargs
+        self,
+        group: dist.ProcessGroup,
+        cuda_init_mode: CUDAInitMode,
+        add_bn: bool,
+        deterministic: bool,
     ):
         super().__init__()
         self.rank = group.rank()
         self.world_size = group.size()
-        torch.manual_seed(0)  # keep everything deterministic
-        assert (
-            d_vocab >= 12
-        ), "dim of vocab should be larger than 12, as we use torch.arange(12) as input"
+        if deterministic:
+            torch.manual_seed(0)
+        d_vocab = 23
```

Comment:
I removed these from the method arguments because no caller ever passes them, so they always take their default values of 23 and 16.